lib/spack/llnl/util/targets.json

```diff
@@ -0,0 +1,655 @@
+{
+  "x86": {
+    "from": null,
```

Comment:
We left the `null` explicitly in because it simplifies searching for generic architectures in the JSON file (they are the only ones with no parents).

Reply:
I left this one in as, much alike `vendor`, this part of the JSON will be more structured if the field is there for each entry (and more easy to work with for external libraries). Let me know if you strongly prefer otherwise.

Reply:
@tgamblin Being a minor issue, I'll mark this one as resolved. Please reopen if you don't consider it so.